app/servers/main/js/favorites.js

```diff
@@ -29,7 +30,17 @@ var favorites = {
   },
 
   _save: function() {
-    asyncStorage.setItem(this.KEYNAME, this.items);
+    var deferred = Promises.defer();
+    asyncStorage.setItem(
+      this.KEYNAME,
+      this.items,
+      () => deferred.resolve(),
+      (e) => {
+        console.log(e);
+        deferred.reject('Cannot save the favorites');
```

Comment:
Do you mind using some |const kErrors| at the top of the files so it will contains all the possible errors instead of having them living alone into the file ?